src/clr-angular/utils/drag-and-drop/droppable/droppable.ts

```diff
@@ -0,0 +1,205 @@
+        }
+        this._isDraggableOver = value;
+    }
+
+    private _group: string|string[];
+    private _dropTolerance: ClrDropTolerance = {top: 0, right: 0, bottom: 0, left: 0};
+
+    @Input("clrGroup")
+    set group(value: string|string[]) {
+        this._group = value;
+    }
+
+    @Input("clrDropTolerance")
+    set dropTolerance(value: number|ClrDropTolerance) {
+        if (typeof value === "number") {
+            this._dropTolerance.top = value;
+            this._dropTolerance.right = value;
+            this._dropTolerance.bottom = value;
+            this._dropTolerance.left = value;
+        } else if (value) {
+            if (value.top) {
+                this._dropTolerance.top = value.top;
+            }
+            if (value.right) {
+                this._dropTolerance.right = value.right;
+            }
+            if (value.bottom) {
+                this._dropTolerance.bottom = value.bottom;
+            }
+            if (value.left) {
+                this._dropTolerance.left = value.left;
+            }
+        }
+    }
+
+    @Output("clrDragStart") dragStartEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+    @Output("clrDragMove") dragMoveEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+    @Output("clrDragEnd") dragEndEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+    @Output("clrDragLeave") dragLeaveEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+    @Output("clrDragEnter") dragEnterEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+    @Output("clrDrop") dropEmitter: EventEmitter<ClrDragEvent<T>> = new EventEmitter();
+
+    private unsubscribeFrom(subscription: Subscription): void {
+        if (subscription) {
+            subscription.unsubscribe();
+        }
+    }
+
+    private checkGroupMatch(draggableGroup: string|string[]): boolean {
+        if (!draggableGroup && this._group) {
+            return false;
+        }
+
+        if (!this._group && draggableGroup) {
+            return false;
+        }
+
+        if (!this._group && !draggableGroup) {
+            return true;
+        }
+
+        if (typeof draggableGroup === "string") {
+            if (typeof this._group === "string") {
+                return this._group === draggableGroup;
+            } else {
+                return this._group.indexOf(draggableGroup) > -1;
+            }
+        } else {
+            if (typeof this._group === "string") {
+                return draggableGroup.indexOf(this._group) > -1;
+            } else {
+                return (this._group as string[]).filter(groupKey => draggableGroup.indexOf(groupKey) > -1).length > 0;
+            }
+        }
+    }
+
+    private isInDropArea(point: {pageX: number; pageY: number}): boolean {
+        if (!point) {
+            return false;
+        }
+
+        if (!this.clientRect) {
+            this.clientRect = this.domAdapter.clientRect(this.droppableEl);
```

Comment:
I think there's a bug here. If I drag an element, stop dragging, then something changes on my page (content added, deleted, ...) `this.clientRect` will still be set to the old one and the computations will be off.

Reply:
@youdz I delete `this.clientRect` in the `onDragEnd` method. And `this.clientRect` get reassigned newly every time `dragMove` event is fired for the first time. Or am I missing something or misunderstanding what you are saying here?

Reply:
Yes, you understood what I was saying. I missed the `delete` in `dragEnd`, you're right.
We might still have the issue I described if the page updates **while I'm dragging**, but I'm okay if that's something we don't want to support. Moving objects around as a user is dragging is probably bad practice for a11y.